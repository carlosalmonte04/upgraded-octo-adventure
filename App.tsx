// In App.js in a new project

import * as React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';

import testIds from 'src/test-ids';
import {signin as strings} from 'src/locales';

import {View, Text, TextInput, Button} from './src/components/common';

export function HomeScreen() {
  const [username, setUsername] = React.useState('');
  const [password, setPassword] = React.useState('');

  return (
    <View>
      <Text testID={testIds.page.signin.usernameLabel}>{strings.username}</Text>
      <TextInput
        testID={testIds.page.signin.usernameInput}
        value={username}
        onChangeText={setUsername}
      />
      <Text testID={testIds.page.signin.passwordLabel}>{strings.password}</Text>
      <TextInput
        testID={testIds.page.signin.passwordInput}
        value={password}
        onChangeText={setPassword}
      />
      <Button
        testID={testIds.page.signin.signinBtn}
        onPress={() => {}}
        title={strings.submit}
        accessibilityLabel={strings.submit}
      />
    </View>
  );
}

export function DetailsScreen() {
  return (
    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
      <Text>Details Screen</Text>
    </View>
  );
}

const Stack = createNativeStackNavigator();

function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator initialRouteName="Home">
        <Stack.Screen name="Home" component={HomeScreen} />
        <Stack.Screen name="Details" component={DetailsScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}

export default App;
