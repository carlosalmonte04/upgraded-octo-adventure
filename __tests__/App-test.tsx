/**
 * @format
 */

import 'react-native';
import React from 'react';
import App, {HomeScreen, DetailsScreen} from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders App correctly', () => {
  renderer.create(<App />);
});

it('renders Home correctly', () => {
  const tree = renderer.create(<HomeScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Details correctly', () => {
  const tree = renderer.create(<DetailsScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
