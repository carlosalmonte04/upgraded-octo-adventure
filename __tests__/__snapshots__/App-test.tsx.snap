// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Details correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Details Screen
  </Text>
</View>
`;

exports[`renders Home correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Home Screen
  </Text>
</View>
`;
